presence/rust/engine/src/ffi.rs

```diff
@@ -0,0 +1,105 @@
+use libc::size_t;
+use std::ptr;
+
+/// Struct to hold an action, identity type and their associated discovery condition.
+#[derive(Clone, Copy, Debug)]
+#[repr(C)]
+pub struct DiscoveryCondition {
+    pub action: i32,
```

Comment:
Please check the size of the action. By the spec, probably it should be a u8.

Reply:
> Please check the size of the action. By the spec, probably it should be a u8.

The spec says 1-3 bytes length. We are not defining wiring format here. As long as the size is larger size the wiring format length, it should work out. I changed to u32.

https://docs.google.com/document/d/1sGA-mN8NN9F07HM9VOvG9olJoIiY1C9ofiRhVZWXFos/edit?resourcekey=0-AWdRnwef3t-n9K2dww5r9A&tab=t.0#bookmark=kix.yoev332ey47k